compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -2650,11 +2650,14 @@ private boolean checkFieldEquivalency(BRecordType lhsType, BRecordType rhsType,
         for (BField lhsField : lhsType.fields.values()) {
             BField rhsField = rhsFields.get(lhsField.name.value);
 
-            // There should be a corresponding RHS field
+            // If LHS field is required, there should be a corresponding RHS field
             if (rhsField == null) {
-                return false;
+                if (Symbols.isOptional(lhsField.symbol)) {
```

Comment:
We have to introduce these changes in the runtime type checker also.

The type test below returns false atm.

```ballerina
type Person record {|
    string name;
    int id;
    float avg?;
|};

type Student record {|
    string name;
    int id;
|};

public function main() {
    Student st = {name: "Amy", id: 1234};

    Person pr = st; // Works.

    boolean res = <any> st is Person; // Evaluates to `false`
}
```

Reply:
Fixed it.